AzureMonitorAgent/ama_tst/modules/helpers.py

```diff
@@ -203,9 +204,66 @@ def run_cmd_output(cmd):
 
 
 #TODO: parse /etc/opt/microsoft/azuremonitoragent/config-cache/configchunks.*.json for workspace ID and VM region
-def find_wkspc_id():
-    return None
+def find_dcr_workspace():
+    global general_info
+    
+    if 'DCR_WORKSPACE_ID' in general_info and 'DCR_REGION' in general_info:
+        return (general_info['DCR_WORKSPACE_ID'], general_info['DCR_REGION'], None)
+    dir_path = '/etc/opt/microsoft/azuremonitoragent/config-cache/configchunks'
+    dcr_workspace = set()
+    dcr_region = set()
+    me_region = set()
+    general_info['URL_SUFFIX'] = '.com'
+    try:
+        for file in os.listdir(dir_path):
+            file_path = dir_path + "/" + file
+            with open(file_path) as f:
+                result = json.load(f)
+                channels = result['channels']
+                for channel in channels:
+                    if channel['protocol'] == 'ods':
+                        # parse dcr workspace id
+                        endpoint_url = channel['endpoint']
+                        worspace_id = endpoint_url.split('https://')[1].split('.ods')[0]
+                        dcr_workspace.add(worspace_id)
+                        # parse dcr region
+                        token_endpoint_uri = channel['tokenEndpointUri']
+                        region = token_endpoint_uri.split('Location=')[1].split('&')[0]
+                        dcr_region.add(region)
+                        # parse url suffix
+                        if '.us' in endpoint_url:
+                            general_info['URL_SUFFIX'] = '.us'
+                        if '.cn' in endpoint_url:
+                            general_info['URL_SUFFIX'] = '.cn'                            
+                    if channel['protocol'] == 'me':
+                        # parse ME region
+                        endpoint_url = channel['endpoint']
+                        region = endpoint_url.split('https://')[1].split('.monitoring')[0]
+                        me_region.add(region)
+    except (FileNotFoundError, AttributeError, KeyError) as e:
+        return (None, None, e)
+
+    general_info['DCR_WORKSPACE_ID'] = dcr_workspace
+    general_info['DCR_REGION'] = dcr_region
+    general_info['ME_REGION'] = me_region
+    return (dcr_workspace, dcr_region, None)
 
 
 def find_vm_region():
     return None
+
+def is_metrics_configured():
+    global general_info
+    if 'metrics' in general_info:
+        return general_info['metrics']
+    
+    metrics_file = "/etc/opt/microsoft/azuremonitoragent/config-cache/metricCounters.json"
+    with open(metrics_file) as f:
+        output = f.read(2)
+        if output != '[]':
```

Comment:
If no metrics is collecting, the content inside this file will be` '[]'`. I checked `if os.path.isfile(<filepath>)` somewhere else with a ERR message. Because the return type seems inconsistent, so I didn't put file existence check in helpers.py

Reply:
Sounds good! Thank you for clarifying!